src/enrichment/query/cosmos/utils.ts

```diff
@@ -24,3 +27,54 @@ export const findByKey = (
     ),
     TE.map(resp => pipe(resp.resource, O.fromNullable))
   );
+
+export const getQuery = (
+  containerName: string,
+  id: string,
+  versionFieldName: string,
+  versionFieldValue: string,
+  partitionKeyField: string,
+  partitionKeyValue: string
+): SqlQuerySpec => ({
+  parameters: [
+    { name: `@${versionFieldName}`, value: `${versionFieldValue}` },
+    { name: `@id`, value: `${id}` },
+    { name: `@${partitionKeyField}`, value: `${partitionKeyValue}` }
+  ],
+  query: `SELECT TOP 1 * FROM ${containerName} f WHERE  f.id = @id AND f.${versionFieldName} = @${versionFieldName} AND f.${partitionKeyField} = @${partitionKeyField} ORDER BY f.${versionFieldName} DESC`
+});
+
+export const findLastVersionByKey = (
+  client: CosmosClient,
+  database: string,
+  containerName: string,
+  versionFieldName: string,
+  versionFieldValue: string,
+  id: string,
+  partitionKeyField: string,
+  partitionKeyValue: string
+): TE.TaskEither<Error, ReadonlyArray<unknown>> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        client
+          .database(database)
+          .container(containerName)
+          .items.query(
+            getQuery(
+              containerName,
+              id,
+              versionFieldName,
+              versionFieldValue,
+              partitionKeyField,
+              partitionKeyValue
+            )
+          )
+          .fetchAll(),
+      () =>
+        new Error(
+          `Impossible to get last version of the item ${id} from container ${containerName}`
+        )
+    ),
+    TE.map(resp => resp.resources)
```

Comment:
by importing RA from ReadonlyArray `import * as RA from "fp-ts/ReadonlyArray";`
```suggestion
    TE.map(resp => resp.resources),
    TE.map(RA.head)
```